imap_processing/lo/l0/lo_science.py

```diff
@@ -439,3 +439,47 @@ def find_valid_groups(
     ]
     valid_groups = [is_sequential(seq_ctrs) for seq_ctrs in grouped_seq_ctrs]
     return valid_groups
+
+
+def organize_spin_data(dataset: xr.Dataset) -> xr.Dataset:
+    """
+    Organize the spin data for Lo.
+
+    The spin data is spread across 28 fields. This function
+    combines each of those fields into 2D arrays for each
+    epoch and spin.
+
+    Parameters
+    ----------
+    dataset : xr.Dataset
+        Lo spin data from packets_to_dataset function.
+
+    Returns
+    -------
+    dataset : xr.Dataset
+        Updated dataset with the spin data organized.
+    """
+    # Get the spin data fields
+    spin_fields = [
+        "start_sec_spin",
+        "start_subsec_spin",
+        "esa_neg_dac_spin",
+        "esa_pos_dac_spin",
+        "valid_period_spin",
+        "valid_phase_spin",
+        "period_source_spin",
+    ]
+
+    for spin_field in spin_fields:
+        packet_fields = [f"{spin_field}_{i}" for i in range(1, 29)]
+        # Combine the spin data fields along a new dimension
+        combined_spin_data = xr.concat(
+            [dataset[field] for field in packet_fields], dim="spin"
```

Comment:
Can you do this directly rather than creating an intermediate by passing in the fields to the data_vars keyword?

```suggestion
            dataset, data_vars=packet_fields, dim="spin"
```

Reply:
Is this what you were thinking?

```python
combined_spin_data = xr.concat(dataset, data_vars=packet_fields, dim="spin")
```

It doesn't look like that will work. This appears to work if you have multiple datasets and want to combine data vars named the same across those two datasets, but it doesn't look like it works in this scenario.

Reply:
Yes, that is what I was thinking. Thanks for checking and sorry for sending you on that incorrect chase!